src/simulators/unitary/unitary_state_chunk.hpp

```diff
@@ -355,32 +334,61 @@ auto State<unitary_matrix_t>::move_to_matrix()
   if(BaseState::num_global_chunks_ == 1){
     return BaseState::qregs_[0].move_to_matrix();
   }
-  else{
-    int_t iChunk;
-    auto state = BaseState::qregs_[0].vector();   //using vector to gather distributed matrix
+  int_t iChunk;
```

Comment:
Code below seems very similar to the one in  `densitymatrix_state_chunk.hpp`, so if you change things there due to comments remember to change here the things that apply.
Also, if it is possible we could refactor the code to avoid the code duplication.

Reply:
Previous codes are quite wrong, so I totally rewrote this section. 

The code here is not same as `DensityMatrixChunk::apply_to_matrix` because the ways of distribution of states to chunks are different. Unitary distributes states in 1-dimension (same as Statevector) but DensityMatrix distributes in 2-dimensions.